src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -35,22 +40,16 @@ internal ResultProvider(Formatter formatter)
             this.Formatter = formatter;
         }
 
-        DkmEvaluationResult IDkmClrResultProvider.GetResult(DkmClrValue value, ReadOnlyCollection<string> formatSpecifiers, string resultName, string resultFullName)
-        {
-            var declaredType = value.DeclaredType; // See #1099981
-            return GetResult(value, declaredType, formatSpecifiers, resultName, resultFullName);
-        }
-
-        internal DkmEvaluationResult GetResult(DkmClrValue value, DkmClrType declaredType, ReadOnlyCollection<string> formatSpecifiers, string resultName, string resultFullName)
+        void IDkmClrResultProvider.GetResult(DkmClrValue value, DkmWorkList workList, DkmClrType declaredType, DkmInspectionContext inspectionContext, ReadOnlyCollection<string> formatSpecifiers, string resultName, string resultFullName, DkmCompletionRoutine<DkmEvaluationAsyncResult> completionRoutine)
         {
             // TODO: Use full name
             try
             {
-                return GetRootResult(value, declaredType, resultName);
+                GetRootResultAsync(value, workList, declaredType, inspectionContext, resultName, result => completionRoutine(new DkmEvaluationAsyncResult(result)));
             }
-            catch (Exception e) when (ExpressionEvaluatorFatalError.CrashIfFailFastEnabled(e))
+            catch (Exception e) when (ExpressionEvaluatorFatalError.ReportNonFatalException(e, DkmComponentManager.ReportCurrentNonFatalException))
             {
-                throw ExceptionUtilities.Unreachable;
+                completionRoutine(DkmEvaluationAsyncResult.CreateErrorResult(e));
```

Comment:
Are we testing this case? That is are we testing that we call 'completionRoutine' when an exception has been thrown?